### PR TITLE
Fix mypy errors in pytorch_lightning/cli.py

### PR DESCRIPTION
**Pull request by mauvilsa**
## What does this PR do?

Fix mypy errors, probably related to a newer version of mypy.

Fixes #14584

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/cli.py [hunk 1]
```diff
@@ -436,6 +436,7 @@ def subcommands() -> Dict[str, Set[str]]:
 
     def _add_subcommands(self, parser: LightningArgumentParser, **kwargs: Any) -> None:
         """Adds subcommands to the input parser."""
+        self.subcommand_parsers: Dict[str, LightningArgumentParser] = {}
```
Comment (akihironitta) [inline src/pytorch_lightning/cli.py:439]: Could this be private?

Comment (carmocca) [reply src/pytorch_lightning/cli.py:439]: I agree with adding the underscore prefix

**Review by carmocca — APPROVED** [review]
Great!

File: src/pytorch_lightning/cli.py (hunk 1 above)
Comment (mauvilsa) [reply src/pytorch_lightning/cli.py]: I added the underscore prefix.

**Review by awaelchli — APPROVED** [review]

**Pull request merged**